### DISCUSSION TITLE
Support for interactive diagrams/flowcharts

### DISCUSSION THREAD

**Discussion by gexahedron** [zauberzeug/nicegui#764] [Ideas / Feature Requests]
### Description

Hi! I know there's a support for mermaid diagrams, but AFAIU they are not interactive (e. g., you can't draw new edges or create new vertices, or delete already existing vertices or edges).
There's also this Vue Flow project - https://github.com/bcakmakoglu/vue-flow - which is interactive, but I don't understand, how to reuse it with NiceGUI. Is it possible right now? They also have an example for Quasar, if that's helpful - https://github.com/bcakmakoglu/vue-flow/tree/master/examples/quasar

**Comment by falkoschindler** [comment]
Hi @gexahedron! That definitely seems like a valuable addition to NiceGUI that could benefit our users.

At this time, we cannot guarantee a timeline for implementing this feature. However, we greatly appreciate and encourage community involvement in the form of API design suggestions, proof-of-concept implementations, or even pull requests that could help bring this feature to life.

**Reply by rodja** [reply]
Indeed. This looks super cool. Please help us to bring this to NiceGUI!!!

**Comment by jakubda1** [comment]
Hi, any Update on this topic? Its 2 years dormant. I thought of using niceGUI for my project, but not having interactive flowchart is one of main features in my project.
Do you need a hand?

**Reply by falkoschindler** [reply]
No, no update, unfortunately.
If you or someone else could create a proof of concept, including the library with some `ui.add_head_html` and a custom `Flow` class, we can estimate how well it could be integrated into the library.
